common/taskana-common/src/main/java/pro/taskana/common/internal/configuration/TaskanaConfigurationInitializer.java

```diff
@@ -0,0 +1,210 @@
+      } catch (Throwable t) {
+        LOGGER.warn(
+            "Could not parse property {} ({}). Using default. Exception: {}",
+            key,
+            property,
+            t.getMessage());
+      }
+    }
+    return Optional.empty();
+  }
+
+  static List<String> splitStringAndTrimElements(String str, String separator) {
+    return splitStringAndTrimElements(str, separator, UnaryOperator.identity());
+  }
+
+  static List<String> splitStringAndTrimElements(
+      String str, String separator, UnaryOperator<String> modifier) {
+    return Arrays.stream(str.split(Pattern.quote(separator)))
+        .filter(s -> !s.isEmpty())
+        .map(String::trim)
+        .map(modifier)
+        .collect(Collectors.toList());
+  }
+
+  static CustomHoliday createCustomHolidayFromPropsEntry(String customHolidayEntry)
+      throws WrongCustomHolidayFormatException {
+    List<String> parts =
+        splitStringAndTrimElements(customHolidayEntry, TASKANA_CUSTOM_HOLIDAY_DAY_MONTH_SEPARATOR);
+    if (parts.size() == 2) {
+      return CustomHoliday.of(Integer.valueOf(parts.get(0)), Integer.valueOf(parts.get(1)));
+    }
+    throw new WrongCustomHolidayFormatException(customHolidayEntry);
+  }
+
+  public static Map<String, List<String>> configureClassificationCategoriesForType(
+      Properties props, List<String> classificationTypes) {
+    Function<String, List<String>> getClassificationCategoriesForType =
+        type ->
+            parseProperty(
+                    props,
+                    TASKANA_CLASSIFICATION_CATEGORIES_PROPERTY + "." + type.toLowerCase(),
+                    p -> splitStringAndTrimElements(p, ",", String::toUpperCase))
+                .orElseGet(ArrayList::new);
+    return classificationTypes.stream()
+        .map(type -> Pair.of(type, getClassificationCategoriesForType.apply(type)))
+        .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+  }
+
+  public static Map<TaskanaRole, Set<String>> configureRoles(
+      String separator, Properties props, boolean shouldUseLowerCaseForAccessIds) {
+    Function<TaskanaRole, Set<String>> getAccessIdsForRole =
+        role ->
+            new HashSet<>(
+                splitStringAndTrimElements(
+                    props.getProperty(role.getPropertyName().toLowerCase(), ""),
+                    separator,
+                    shouldUseLowerCaseForAccessIds
+                        ? String::toLowerCase
+                        : UnaryOperator.identity()));
+
+    return Arrays.stream(TaskanaRole.values())
+        .map(role -> Pair.of(role, getAccessIdsForRole.apply(role)))
+        .collect(Collectors.toMap(Pair::getLeft, Pair::getRight));
+  }
+
+  public static List<Field> getAllFields(List<Field> fields, Class<?> type) {
+    fields.addAll(Arrays.asList(type.getDeclaredFields()));
+    if (type.getSuperclass() != null) {
+      getAllFields(fields, type.getSuperclass());
+    }
+    return fields;
+  }
```

Comment:
We have this function in `ObjectAttributeChangeDetector`. Maybe create a `ReflectionUtil`class?